clients/githubrepo/graphql.go

```diff
@@ -205,19 +212,42 @@ func (handler *graphqlHandler) setup() error {
 			"issueCommentsToAnalyze": githubv4.Int(issueCommentsToAnalyze),
 			"reviewsToAnalyze":       githubv4.Int(reviewsToAnalyze),
 			"labelsToAnalyze":        githubv4.Int(labelsToAnalyze),
-			"commitsToAnalyze":       githubv4.Int(commitsToAnalyze),
+			"commitsToAnalyze":       githubv4.Int(CommitsToAnalyze),
 			"commitExpression":       githubv4.String(commitExpression),
+			"historyCursor":          (*githubv4.String)(nil),
 		}
-		if err := handler.client.Query(handler.ctx, handler.data, vars); err != nil {
-			handler.errSetup = sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("githubv4.Query: %v", err))
+		if CommitsToAnalyze < 99 {
+			if err := handler.client.Query(handler.ctx, handler.data, vars); err != nil {
+				handler.errSetup = sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("githubv4.Query: %v", err))
+				return
+			}
+			handler.archived = bool(handler.data.Repository.IsArchived)
+			handler.issues = issuesFrom(handler.data)
+			handler.commits, handler.errSetup = commitsFrom(handler.data, handler.repourl.owner, handler.repourl.repo)
 			return
 		}
-		handler.archived = bool(handler.data.Repository.IsArchived)
-		handler.commits, handler.errSetup = commitsFrom(handler.data, handler.repourl.owner, handler.repourl.repo)
-		if handler.errSetup != nil {
-			return
+		var allCommits []clients.Commit
+		commitsLeft := CommitsToAnalyze
+		vars["commitsToAnalyze"] = githubv4.Int(100)
+		for ; commitsLeft > 0; commitsLeft = commitsLeft - 100 {
```

Comment:
I wonder if this loop can take care of the case where `CommitsToAnalyze`, instead of having a special case for it.
Same questions for the other places in the code where it's used.